resources/metadata/helmExecute.yaml

```diff
@@ -0,0 +1,318 @@
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        default: 300
+      - name: helmValues
+        type: "[]string"
+        description: List of helm values as YAML file reference or URL (as per helm parameter description for `-f` / `--values`)
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: image
+        aliases:
+          - name: deployImage
+        type: string
+        description: Full name of the image to be deployed.
+        resourceRef:
+          - name: commonPipelineEnvironment
+            param: container/imageNameTag
+        mandatory: true
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: keepFailedDeployments
+        type: bool
+        description: Defines whether a failed deployment will be purged
+        default: false
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: kubeConfig
+        type: string
+        description: Defines the path to the "kubeconfig" file.
+        scope:
+          - GENERAL
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        secret: true
+        resourceRef:
+          - name: kubeConfigFileCredentialsId
+            type: secret
+          - type: vaultSecretFile
+            name: kubeConfigFileVaultSecretName
+            default: kube-config
+      - name: kubeContext
+        type: string
+        description: Defines the context to use from the "kubeconfig" file.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+      - name: namespace
+        aliases:
+          - name: helmDeploymentNamespace
+        type: string
+        description: Defines the target Kubernetes namespace for the deployment.
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        default: default
+      - name: dockerConfigJSON
+        type: string
+        description: Path to the file `.docker/config.json` - this is typically provided by your CI/CD system. You can find more details about the Docker credentials in the [Docker documentation](https://docs.docker.com/engine/reference/commandline/login/).
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        secret: true
+        resourceRef:
+          - name: dockerConfigJsonCredentialsId
+            type: secret
+          - type: vaultSecretFile
+            name: dockerConfigFileVaultSecretName
+            default: docker-config
+      - name: helmCommand
+        type: string
+        description: "Helm: defines the command `install`, `lint`, `package`, `test`, `upgrade` and etc."
+        scope:
+          - PARAMETERS
+          - STAGES
+          - STEPS
+        mandatory: true
+        possibleValues:
+          - upgrade
+          - install
+          - lint
+          - test
+          - uninstall
+          - package
+          - push
+      - name: dryRun
```

Comment:
Not all commands allow `--dryRun`. For example `helm package`. So this might be confusing.

Reply:
it will be thought out